core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -640,8 +640,9 @@ object AdminUtils extends Logging with AdminUtilities {
    * Write out the entity config to zk, if there is any
    */
   private def writeEntityConfig(zkUtils: ZkUtils, entityPath: String, config: Properties) {
-    val map = Map("version" -> 1, "config" -> config.asScala)
-    zkUtils.updatePersistentPath(entityPath, Json.encode(map))
+    val map = Map("version" -> 1, "config" -> config)
+
+    zkUtils.updatePersistentPath(entityPath, Json.encodeToJsonString(map))
```

Comment:
Since this class is deprecated, it can probably continue to use the old `legacy` methods. My concern is that we have moved our code to use `KafkaZkClient` and I'm not sure how good the test coverage for this code is after those changes.

Reply:
Yes that makes sense. I will rollback the changes for this file.